azure/converters/tags.go

```diff
@@ -23,6 +23,10 @@ import (
 
 // MapToTags converts a map[string]*string into a infrav1.Tags.
 func MapToTags(src map[string]*string) infrav1.Tags {
+	if src == nil {
```

Comment:
can you please add a unit test that covers this scenario?

Reply:
@CecileRobertMichon I now added unit tests for `MapToTags` and `TagsToMap`. Unit tests were missing completely for tag converters if I looked in the correct files, so I created a completely new file. Please review. thanks